components/brave_rewards/resources/extension/brave_rewards/background/api/tabs_api.ts

```diff
@@ -40,6 +50,28 @@ export const onTabData = (tab: chrome.tabs.Tab, activeTabIsLoadingTriggered?: bo
   rewardsPanelActions.onTabRetrieved(tab, activeTabIsLoadingTriggered)
 }
 
+export const processPendingTabs = () => {
```

Comment:
could you please explain this pending tabs flow?

Reply:
Yes, we need two pieces of information when a user visits a Twitter URL:

1. We need the auth headers, so that we can query the Twitter API for the user ID
2. We need the `onTabData` callback to tell us that we've visited a Twitter URL and begin the process of processing the publisher activity

If those two events happen in the order listed above, we're good. If they happen in the opposite order, then we have a problem as without the auth headers we will not be able to retrieve the correct Twitter user ID and then we'll try to process a URL with a missing user ID.

My idea was that if a tab becomes active before we have the Twitter auth headers for it, we insert it into a dictionary and then only process it once the auth headers are received (see related callback in `twitterAuth.ts`). If the user switches away from that tab in the meantime, we'll just discard it. Open to other ideas too, this was the main idea that occurred to me though.

Reply:
make sense, thank you